src/goto-instrument/contracts/utils.h

```diff
@@ -28,6 +29,11 @@ class havoc_if_validt : public havoc_utilst
   {
   }
 
+  void append_havoc_code_for_expr(
+    const source_locationt location,
+    const exprt &expr,
+    goto_programt &dest) const override;
```

Comment:
I added some comments within the functions, but since are short on time, I think we should defer doxygen comments on all of the contracts functions to a separate PR.